src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/DataTransferService.java

```diff
@@ -1,47 +1,210 @@
-    // TODO Implement onRequest workflow logic.
-    initiateBundleRequest(HIPHealthInformationRequest);
+  public void
+  requestOnReceived(HIPHealthInformationRequest hipHealthInformationRequest) {
+    HIPHealthInformationRequestAcknowledgement hipHealthInformationRequestAcknowledgement = null;
+    RespRequest responseRequestId =
+        RespRequest.builder().requestId(hipHealthInformationRequest.getRequestId()).build();
+    String consentId = hipHealthInformationRequest.getHiRequest().getConsent().getId();
+    HiRequest hiRequest =
+        HiRequest.builder()
+            .sessionStatus("ACKNOWLEDGED")
+            .transactionId(hipHealthInformationRequest.getTransactionId())
+            .build();
+    if (logsRepo.findByConsentId(consentId) != null) {
+      hipHealthInformationRequestAcknowledgement =
+          HIPHealthInformationRequestAcknowledgement.builder()
+              .requestId(UUID.randomUUID().toString())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .hiRequest(hiRequest)
+              .resp(responseRequestId)
+              .build();
+    } else {
+      errorResponse.setMessage("HIP -> ConsentId not found in cache");
+      errorResponse.setCode(1000);
+      log.error("DataOnRequest body -> making error body, consentId not found in cache");
+      hipHealthInformationRequestAcknowledgement =
+          HIPHealthInformationRequestAcknowledgement.builder()
+              .requestId(UUID.randomUUID().toString())
+              .timestamp(Utils.getCurrentTimeStamp())
+              .error(errorResponse)
+              .resp(responseRequestId)
+              .build();
+    }
+    log.info(hipHealthInformationRequestAcknowledgement.toString());
+    try {
+      ResponseEntity<GatewayGenericResponse> response =
+          requestManager.fetchResponseFromGateway(
+              dataOnRequestPath, hipHealthInformationRequestAcknowledgement);
+      log.debug(dataOnRequestPath + " : dataOnRequest: " + response.getStatusCode());
+      if (response.getStatusCode() == HttpStatus.ACCEPTED) {
+        requestLogService.dataTransferRequest(
+            hipHealthInformationRequest,
+            RequestStatus.DATA_ON_REQUEST_SUCCESS,
+            hipHealthInformationRequestAcknowledgement);
+      } else if (Objects.nonNull(response.getBody())
+          && Objects.nonNull(response.getBody().getErrorResponse())) {
+        requestLogService.dataTransferRequest(
+            hipHealthInformationRequest,
+            RequestStatus.DATA_ON_REQUEST_ERROR,
+            hipHealthInformationRequestAcknowledgement);
+      }
+    } catch (Exception ex) {
+      String error =
+          "Exception while Initiating HIP auth: "
+              + ex.getMessage()
+              + " unwrapped exception: "
+              + Exceptions.unwrap(ex);
+      log.debug(error);
+    }
+
+    //    initiateBundleRequest(HIPHealthInformationRequest);
```

Comment:
This can be initiated only if the `Consent` is in `GRANTED` state. Will this validation handled in next PR?

Reply:
As of HIP, we don't have any changes to the data, only the HIU consent manager needs to change them because the HIU stores data.